caikit/core/data_model/dataobject.py

```diff
@@ -54,8 +52,43 @@
 
 ## Public ######################################################################
 
-# Common package prefix
-CAIKIT_DATA_MODEL = "caikit_data_model"
+
+class _DataObjectBaseMetaClass(_DataBaseMetaClass):
+    """This metaclass is used for the DataObject base class so that all data
+    objects can delay the creation of their proto class until after the
+    metaclass has been instantiated.
+    """
+
+    def __new__(mcs, name, bases, attrs):
+        """When instantiating a new DataObject class, the proto class will not
+        yet have been generated, but the set of fields will be known since the
+        class will be the raw input representation of a @dataclass
+        """
+
+        # Get the annotations that will go into the dataclass
+        if name != "DataObjectBase":
+            raw_field_names = attrs.get("__annotations__")
+            if raw_field_names is None:
+                raise TypeError(
+                    "All DataObjectBase classes must follow dataclass syntax"
+                )
+
+            # TODO: Sort out oneof field names
+            field_names = list(raw_field_names.keys())
+
+            # Add the forward declaration
+            attrs[_DataBaseMetaClass._FWD_DECL_FIELDS] = field_names
+
+        # Delegate to the base metaclass
+        return super().__new__(mcs, name, bases, attrs)
+
+
+class DataObjectBase(DataBase, metaclass=_DataObjectBaseMetaClass):
```

Comment:
This base class serves no purpose other than to bind `DataBase` and `_DataObjectBaseMetaClass` together. In the future, we may need real logic here, but we should try hard to avoid it since we want all classes declared using `@dataobject` to cleanly line up with those that use `DataBase` directly